hpilo.py

```diff
@@ -1605,7 +1605,10 @@ def mod_dir_config(self, dir_authentication_enabled=None,
             if key.endswith('_priv'):
                 if isinstance(val, basestring):
                     val = val.replace('oemhp_', '').replace('_priv', '').split(',')
-                val = ','.join([str(privmap.get(x,x)) for x in val])
+                if isinstance(val, list):
+                    val = ','.join([str(privmap.get(x,x)) for x in val])
```

Comment:
Hmm. This would break if one were to pass a tuple or other iterator instead of a list. Maybe change this to checking whether val is a non-string iterator?

Reply:
will do.

Reply:
If this does not match, the next "if" statement will if val is not a string.